### PR TITLE
Asset management

### PR DESCRIPTION
**Pull request by rherriman**
This seems to work perfectly for loading/storing manifests, default.avarascript files, HULL resources (though I still need to implement/call `AssetManager::ReviewPriorities` for those...), and BSP files. Still need to implement sound loading and preprocessing (combining sample data and sample header data) and related updates to CSoundHub.

### CODE REVIEW

**Comment by rherriman** [comment]
This branch should now have feature parity with main, and all the old (non-resource fork related) functions have been removed from Resource.h.

Audio code updates were more disruptive than I expected but I seem to have smoothed out the initial kinks, which ultimately resulted in a nicer and potentially speedier design.

Memory usage appears to be on par with main, potentially even a tiny bit lower.

These changes also allow level sets to include audio and hull resources from other sets, which was previously not yet possible in main.

File: src/compat/Resource.cpp
Comment (assertivist) [inline src/compat/Resource.cpp:1]: Surely we can nuke the rest of this file with some careful pruning. I'm curious to know what still uses `BundlePath` and `OSType`s

**Review by assertivist — APPROVED** [review]
An incredibly cleanly done work. a vast step forward towards remote level loading. An all around handsome PR.

File: src/game/CAbstractPlayer.cpp [hunk 1]
```diff
@@ -253,9 +253,9 @@ CAbstractActor *CAbstractPlayer::EndScript() {
     loseSound = ReadLongVar(iLoseSound);
     loseVolume = ReadFixedVar(iLoseVolume);
 
-    gHub->PreLoadSample(incarnateSound);
-    gHub->PreLoadSample(winSound);
-    gHub->PreLoadSample(loseSound);
+    auto _ = AssetManager::GetOgg(incarnateSound);
```
[the remark this answers is not in the document]
Comment (assertivist) [reply src/game/CAbstractPlayer.cpp:256]: I wrote this before I realized exactly how many places you put `auto _ =`. I don't really want to send you chasing all those down, i can get over myself

**Pull request merged**